angular-material-calendar.js

```diff
@@ -28,11 +28,33 @@ angular.module("materialCalendar").service("materialCalendar.Calendar", [functio
             return this.weekStartsOn;
         };
 
+        this.setStartDate = function (i) {
+            var d = parseInt(i || 1, 10);
+            if (!isNaN(d) && d >= 1 && d <= 31) {
+                this.startDate = d;
+            } else {
+                this.startDate = 1;
+            }
+            return this.startDate;
+        };
+
+        this.setNoOfDays = function (i) {
+            var d = parseInt(i || 0, 10);
+            if (!isNaN(d) && d > 0 ) {
+                this.noOfDays = d;
+            } else {
+                this.noOfDays = 0;
+            }
+            return this.noOfDays;
+        };
+
         this.options = angular.isObject(options) ? options : {};
         this.year = now.getFullYear();
         this.month = now.getMonth();
         this.weeks = [];
         this.weekStartsOn = this.setWeekStartsOn(this.options.weekStartsOn);
+        this.startDate = this.setStartDate(this.options.startDate);
```

Comment:
Perhaps this is better called startDayOfMonth? startDate suggest a value like '2016-01-31'